Job/Product.php

```diff
@@ -2120,11 +2122,28 @@ public function importMedia()
                 ];
                 $connection->insertOnDuplicate($galleryEntityTable, $data, array_keys($data));
 
+                if ($image['position']) {
+                    $recordId = $connection->fetchOne(
+                        $connection->select()
+                            ->from($galleryValueTable, ['record_id'])
+                            ->where('`value_id` = ?', $valueId)
+                            ->where('`entity_id` = ?', $row[$columnIdentifier])
+                    );
+
+                    $data['position'] = $image['position'];
+
+                    if ($recordId) {
```

Comment:
insertOnDuplicate is not possible here since `value_id` and `entity_id` are ambiguous